src/core/cgroup.c

```diff
@@ -4040,10 +4040,11 @@ int unit_get_memory_current(Unit *u, uint64_t *ret) {
         return cg_get_attribute_as_uint64("memory", u->cgroup_path, r > 0 ? "memory.current" : "memory.usage_in_bytes", ret);
 }
 
-static int unit_get_memory_peak_raw(Unit *u, uint64_t *ret) {
+static int unit_get_memory_attr_raw(Unit *u, const char* mem_attribute, uint64_t *ret) {
```

Comment:
i like the unification effort a log here, but I think it might be nicer to follow the scheme we followed in unit_get_io_accounting_raw(), i.e. add a new enum for the various memory metrics, plus a table to map them to the relevant attribute field.

but also doesn't matter much, we can do this another time